pkg/netpol/diff/diff.go

```diff
@@ -461,18 +472,32 @@ func getFormatter(format string) (diffFormatter, error) {
 	}
 }
 
+// RemovedConnsPeers encapsulates a removed connection between two peers and indications if any of the peers was removed
+type RemovedConnsPeers struct {
+	removedConn connlist.Peer2PeerConnection
+	removedSrc  bool
+	removedDst  bool
+}
+
+// AddedConnsPeers encapsulates an added connection between two peers and indications if any of the peers is new
+type AddedConnsPeers struct {
+	addedConn connlist.Peer2PeerConnection
+	addedSrc  bool
+	addedDst  bool
+}
+
```

Comment:
will it simplify if you remove those new types and just use the type `ConnsPair`,  with additional fields for that type that indicate if it is added/removed/changed, including src/dst info whether it is added/removed?

Reply:
regarding using the `ConnsPair `type for the 3 diff types:
-> in commit [e6ceb4db](https://github.com/np-guard/netpol-analyzer/pull/202/commits/e6ceb4db15ea5d56a06d514bccb2678de433ce6c):
I chose to write the `connectivityDiff struct` , this way :
```
type connectivityDiff struct {
	diffConns []*ConnsPair
```
since all 3 diff types (changed, removed and added) are represented with `[]*ConnsPair ` , and single `ConnsPair ` object contains also the diffType.
however, in order to implement the `ConnectivityDiff interface`, I had to add the func : 
`func (c *connectivityDiff) diffConnectionsByType(diffType string) []*ConnsPair`

-> But, I hesitated if this was efficient or not, so in commit [2fd4cb8f](https://github.com/np-guard/netpol-analyzer/pull/202/commits/2fd4cb8fd3b1832135cf1f4de32dad479e2fe430) 
I chose to return the struct to be as following (the original with the new types for added and removed):
```
type connectivityDiff struct {
	removedConns []*ConnsPair
	addedConns   []*ConnsPair
	changedConns []*ConnsPair
}
```
without any additional funcs to implement the `ConnectivityDiff interface` (and even though the 3 elements of the struct has same type)